indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -28,8 +28,22 @@
             openInDialog: false,
             // The title for the category navigator dialog
             dialogTitle: $T.gettext("Select a category"),
-            // Restrict action to categories with no subcategories
-            selectLeafOnly: false,
+            // Allow action to specific categories
+            actionOn: {
+                categoriesWithSubcategories: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select a category that contains subcategories")
+                },
+                categoriesWithEvents: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select a category that contains events")
+                },
+                self: {
+                    disabled: false,
+                    message: $T.gettext("You cannot select the same category"),
```

Comment:
It's not really "current" if for example you choose many subcategories and perform a bulk operation. The current would apply only when you try to move the category you are browsing. It's more like "You cannot select the category you selected to be moved"...